### PR TITLE
[Enhancement] optimize lambda functions by rewriting replicate() and splitting large chunks 

### PR DESCRIPTION
**Pull request by fzhedu**
Signed-off-by: fzhedu <fzhedu@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

### problem1:
align captured column with array's offsets, which copy one captured value mulitple times  equalling to array's, for example: 
captuerd value =1, array=['a','b','c'], then captured value 1 is replicated 3 times to [1,1,1], aligning to array's size.

this replicate takes too much time, so we optimized it to write specific replicat() for FixedLengthColumn, BinaryColumn and ConstColumn.

### problem2:

as lambda functions adopt array->elements to compute, but array->elements's szie may be vary large and various during executing, so split the array->elements into small chunks to evaluate, however which induces too much copy cost. we should to avoid the expensive copy cost later. 

<byte-sheet-html-origin data-id="RKqHJKmRoO-1663660075844" data-version="3" data-is-embed="false" data-importRangeRawData-spreadSource="https://starrocks.feishu.cn/sheets/shtcndJJCISGcsb70prBNBOt37c" data-importRangeRawData-range="&#39;Sheet6&#39;!C2:E3">

 sql |  before | after
-- | -- | --
SELECT max(g), max(g0), max(g1), max(g2), max(g3), max(g4), max(g5), max(g6), max(g7), max(g8), max(g9) FROM(SELECT array_sum(array_map(x->x * x + x*b, a)) g, array_sum(array_map((x, y)->x * x + x*b - b * y + 10, a, a)) g0, array_sum(array_map((x, y)->x * x + x*b - b * y + 11, a, a)) g1, array_sum(array_map((x, y)->x * x + x*b - b * y + 12, a, a)) g2, array_sum(array_map((x, y)->x*x + x*b - b * y + 13, a, a)) g3, array_sum(array_map((x, y)->x * x + x*b - b * y + 14, a, a)) g4, array_sum(array_map((x, y)->x*x + x*b - b * y + 15, a, a)) g5, array_sum(array_map((x, y)->x * x + x*b - b * y + 16, a, a)) g6, array_sum(array_map((x, y)->x*x + x*b - b * y + 17, a, a)) g7, array_sum(array_map((x, y)->x * x + x*b - b * y + 18, a, a)) g8, array_sum(array_map((x, y)->x*x + x*b - b * y + 19, a, a)) g9 FROM (SELECT aint a, lo_custkey b FROM tarray)A)B; | 17.95 | 13.36

</byte-sheet-html-origin>

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/column/column.h [hunk 1]
```diff
@@ -152,6 +152,17 @@ class Column {
 
     virtual void append(const Column& src) { append(src, 0, src.size()); }
 
+    // TODO(fzh): optimize replicate() for specific columns, such as array_column, object_column.
+    virtual ColumnPtr replicate(const std::vector<uint32_t>& offsets) {
+        auto dest = this->clone_empty();
+        auto dest_size = offsets.size() - 1;
+        DCHECK(this->size() >= dest_size) << "The size of the source column is less when duplicating it.";
+        dest->reserve(offsets.back());
+        for (int i = 0; i < offsets.size() - 1; ++i) {
```
Comment (imay) [inline be/src/column/column.h:161]: why not use dest_size directly?

File: be/src/column/column.h [hunk 2]
```diff
@@ -152,6 +152,17 @@ class Column {
 
     virtual void append(const Column& src) { append(src, 0, src.size()); }
 
+    // TODO(fzh): optimize replicate() for specific columns, such as array_column, object_column.
+    virtual ColumnPtr replicate(const std::vector<uint32_t>& offsets) {
```
Comment (imay) [inline be/src/column/column.h:156]: should add some comments to explain what this function is for.

File: be/src/column/column.h (hunk 1 above)
Comment (fzhedu) [reply be/src/column/column.h]: fixed

File: be/src/column/column.h (hunk 2 above)
Comment (fzhedu) [reply be/src/column/column.h:159]: added, to aligned with array's offset in lambda functions.

**Review by kangkaisen — DISMISSED** [review]

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by ZiheLiu — APPROVED** [review]

**Pull request merged**